msal/samples/desktop/SampleApp/MsalAuthHelper.cs

```
@@ -37,54 +38,67 @@ namespace SampleApp
 {
     class MsalAuthHelper
     {
-        private string _clientId;
+        private readonly string _clientId;
+        private readonly string user = ""; //can be empty for IWA and U/P
 
         public PublicClientApplication Application { get; private set; }
 
         public MsalAuthHelper(string clientId)
         {
             _clientId = clientId;
-            Application = new PublicClientApplication(_clientId, "https://login.microsoftonline.com/common/",
+            Application = new PublicClientApplication(_clientId, "https://login.microsoftonline.com/organizations/",
                 CachePersistence.GetUserCache());
         }
 
-        public async Task<IAccount> SignInAsync()
+        public async Task<string> GetTokenForCurrentAccountAsync(IEnumerable<string> scopes, IAccount account)
         {
+            AuthenticationResult result = null;
+            Exception exception = null;
             try
             {
-                AuthenticationResult result = await Application.AcquireTokenAsync(new[] {"user.read", "calendars.read"}).ConfigureAwait(false);
-                return result.Account;
+                result = await Application.AcquireTokenSilentAsync(scopes, account).ConfigureAwait(false);
+                return result.AccessToken;
             }
-            catch (Exception exc)
+            catch (MsalUiRequiredException)
             {
-                MessageBox.Show(exc.Message, "Sign in failed", MessageBoxButtons.OK, MessageBoxIcon.Error);
+                try
+                {
+                    result = await Application.AcquireTokenAsync(scopes).ConfigureAwait(false);
+                    return result.AccessToken;
+                }
+                catch (MsalServiceException ex)
```

Not needed since you catch all exceptions at line 74